core/chains/terra/terratxm/models.go

```diff
@@ -0,0 +1,78 @@
+package terratxm
+
+import (
+	"time"
+
+	"github.com/smartcontractkit/chainlink-terra/pkg/terra/client"
+
+	"github.com/smartcontractkit/terra.go/msg"
+)
+
+// State represents the state of a given terra msg
+// Happy path: Unstarted->Broadcasted->Confirmed
+type State string
+
+var (
+	// Unstarted means queued but not processed.
+	// Valid next states: Broadcasted, Errored (sim fails)
+	Unstarted State = "unstarted"
+	// Broadcasted means included in the mempool of a node.
+	// Valid next states: Confirmed (found onchain), Errored (tx expired waiting for confirmation)
+	Broadcasted State = "broadcasted"
+	// Confirmed means we're able to retrieve the txhash of the tx which broadcasted the msg.
+	// Valid next states: none, terminal state
+	Confirmed State = "confirmed"
+	// Errored means the msg reverted in simulation OR the tx containing the message timed out waiting to be confirmed
+	// TODO: when we add gas bumping, we'll address that timeout case
+	// Valid next states, none, terminal state
+	Errored State = "errored"
+)
+
+// TerraMsg a terra msg
+type TerraMsg struct {
+	ID         int64
+	ContractID string
+	State      State
+	Msg        []byte
+	TxHash     *string
```

Comment:
Consider using `null.String`

Reply:
hmm with the move to sqlx I'm not sure what that buys us, both db and json marshaling work fine with just a pointer. Perhaps less prone to nil pointer panics, but I think in both cases you'd be doing a check

Reply:
I suppose I'm just partial to explicit maybe types, but as you say it doesn't make too much difference